csrc/dynamic_transform.cpp

```diff
@@ -0,0 +1,510 @@
+ private:
+  ExpressionEvaluator* expr_eval_ = nullptr;
+
+  DynamicTransformInfo info_;
+};
+
+bool DynamicTransformInfo::operator==(const DynamicTransformInfo& other) const {
+  if (this == &other) {
+    return true;
+  }
+
+  if (fusion_ != other.fusion_) {
+    return false;
+  }
+
+  if (reshape_transforms_.size() != other.reshape_transforms_.size()) {
+    return false;
+  }
+
+  for (const auto i : c10::irange(reshape_transforms_.size())) {
+    const auto& transform = reshape_transforms_.at(i);
+    const auto& other_transform = other.reshape_transforms_.at(i);
+    std::cerr << "This Transform: " << transform.first->toString() << ", "
+              << transform.second.toString() << std::endl;
+    std::cerr << "Other Transform: " << other_transform.first->toString()
+              << ", " << other_transform.second.toString() << std::endl;
+    if (transform != other_transform) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// TODO
+std::string DynamicTransformInfo::toString() const {
+  std::stringstream ss;
+  ss << "DynamicTransformInfo\n";
+  std::string indent = "  ";
+  ss << indent << "Reshape:\n";
+  for (const auto& kv : reshape_transforms_) {
+    ss << indent << indent << kv.first->toString() << ", "
+       << kv.second.toString() << "\n";
+  }
+  return ss.str();
+}
+
+void DynamicTransformInfoBuilder::augmentExprEvaluator() {
+  const auto mapped_sets = ExactRootDomainMap(info_.fusion()).getMappedSets();
+
+  // std::cerr << "Augmenting ExprEval\n";
+
+  for (const auto& set : mapped_sets.disjointSets()) {
+    // std::cerr << "Disjoint set\n";
+    int64_t known_size = -1;
+    std::vector<Val*> unknown_vals;
+    for (const auto id : *set) {
+      // std::cerr << "ID: " << id->toString() << std::endl;
+      auto eval_val = expr_eval_->evaluate(id->extent());
+      if (eval_val.has_value()) {
+        TORCH_INTERNAL_ASSERT(eval_val->isInt(), "Invalid extent value");
+        int64_t this_size = eval_val->as<int64_t>();
+        if (known_size != -1) {
+          TORCH_INTERNAL_ASSERT(
+              known_size == this_size,
+              "Conflicting sizes: ",
+              known_size,
+              ", ",
+              this_size);
+        } else {
+          known_size = this_size;
+        }
+      } else {
+        unknown_vals.push_back(id->extent());
+      }
+    }
+
+    if (known_size == -1 || unknown_vals.empty()) {
+      continue;
+    }
+
+    // Binding unknown vals to known_val
+    for (auto unknown_val : unknown_vals) {
+      // std::cerr << "Augment: " << unknown_val->toString() << " -> "
+      //<< known_size << std::endl;
+      expr_eval_->bind(unknown_val, known_size);
+    }
+  }
+}
+
+void DynamicTransformInfoBuilder::handle(ViewOp* op) {
+  std::cerr << "Reshape: " << op->toString();
+
+  // Determine if this is a dynamic reshape. If the output tv doesn't
+  // have an rfactor domain, no view transform is set up, which means
+  // its output domain is still just a placeholder
```

Comment:
Which should be resolved at this point?

Reply:
Sorry, this comment is stale. Initially I created a dynamic reshape op with no rfactor domain, which isn't the case right now.

At this point, the actual extents of the input and output tensors are available, and we use that information to do the the same view analysis we do for static reshapes.